lib/tasks_private/lib/github_file.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'json'
+require 'base64'
+require 'pathname'
+require 'net/http'
+require_relative 'simple_logging'
+
+class GithubFile
```

Comment:
I think it would be easier to just use Octokit than to build your own.  Though I don't understand the purpose of the cache_dir, nor the commits_uri